src/server/core/ml/tokenizer.py

```diff
@@ -0,0 +1,206 @@
+import re, enchant
+
+# Preprocessing #
+#################
+
+# English dictionary
+en_US = None
+
+# Special characters inserted on preprocessing
+special_characters = [
+    ("{space}", " "),
+    ("{comma}", ","),
+    ("{dot}", "."),
+    ("{colon}", ":")
+]
+
+tags = ["name", "number", "email", "phone", "url", "phrase"]
+
+# Regexes used in preprocessing
+regexes = {
+    # preprocessing regexes
+    "number": r"[0-9]+((\.|,)([0-9]+)+)+",
+    "email": r"[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+",
+    "phone": r"(\(?[0-9]{3,5}\)?)[\s]?([\-\/])?\s?([0-9-]{4,10})(?![0-9])",
+    "url": r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+",
+    "name": r"((Mrs|Ms|Mr|Dr)\.? ?)?((([A-Z][a-z]+)(\-[A-Z][a-z]+)?)|([A-Z](\.|’)))( ?(([A-Z][a-z]+)(\-[A-Z][a-z]+)?))?",
+    #            title                  first name                                 last name
+
+    # processing regexes
+    "tag": r"(<((name)|(number)|(email)|(phone)|(url)|(phrase))>)([^<]+)(<\/((name)|(number)|(email)|(phone)|(url)|(phrase))>)",
+    "punctuation": r"([\.,:;!?'\"“”()]+)?([^\.,:;!?'\"“”()]+)?([\.,:;!?'\"“”()]+)?([^\.,:;!?'\"“”()]+)?",
```

Comment:
same as above, and if this regexp is supposed to match one or more punctuation marks, you can just use [string.punctuation]+